pennylane/ops/qubit.py

```diff
@@ -44,10 +47,15 @@ class Hadamard(Observable, Operation):
     num_wires = 1
     par_domain = None
     eigvals = pauli_eigs(1)
+    matrix = np.array([[INV_SQRT2, INV_SQRT2], [INV_SQRT2, -INV_SQRT2]])
 
-    @staticmethod
-    def _matrix(*params):
-        return np.array([[1, 1], [1, -1]]) / np.sqrt(2)
+    @classmethod
+    def _matrix(cls, *params):
+        return Hadamard.matrix
```

Comment:
Why not `cls.matrix`? Could even avoid having to rewrite this method for every subclass?

Reply:
also, it is might not be the best to go against the sort convention of a public attribute/method depending on a private property/attribute/method since it can lead to confusion (in this case `matrix` depends on `_matrix` in the `Operator` class)

Reply:
First, @co9olguy you're right. In this case `cls.matrix` makes more sense. I actually added this method before I changed `_matrix` to classmethod and didn't update.

@antalszava I was deliberately breaking this abstraction here to avoid duplicate code. For constant gates, it is the fastest to just hardcode `matrix` to be a constant numpy array. This avoids the overhead of `matrix` calling `_matrix` to get the constant array. 
One could imagine a `ConstantOperation` class, but I think that would get out of hand...